### PR TITLE
[SW-14769] shared email toggle

### PR DESCRIPTION
**Pull request by rajush**

### CODE REVIEW

File: widgets/cmEmailAddresses.html [hunk 1]
```diff
@@ -60,11 +60,10 @@
             <div class="cm-email-addresses-left-pane" ng-show="!!contact.emailEditMode">
                 <div class="cm-shared-email">
                     <label class="cm-field-label">Share From Existing Contact:</label>
-                    <md-input-container md-no-float ng-click="!contact.IsShareEmailContactId && !contact.isNewContact ? toggleSharedEmail() : angular.noop();" ng-class="{ 'cm-disabled': !!contact.IsShareEmailContactId || !!contact.isNewContact }">
+                    <md-input-container md-no-float ng-click="toggleSharedEmail();">
```
Comment (srosenbloom) [inline widgets/cmEmailAddresses.html]: Need to keep IsShareEmailCondition here on the ng-click and ng-class

File: widgets/cmEmailAddresses.html [hunk 2]
```diff
@@ -60,11 +60,10 @@
             <div class="cm-email-addresses-left-pane" ng-show="!!contact.emailEditMode">
                 <div class="cm-shared-email">
                     <label class="cm-field-label">Share From Existing Contact:</label>
-                    <md-input-container md-no-float ng-click="!contact.IsShareEmailContactId && !contact.isNewContact ? toggleSharedEmail() : angular.noop();" ng-class="{ 'cm-disabled': !!contact.IsShareEmailContactId || !!contact.isNewContact }">
```
Comment (srosenbloom) [inline widgets/cmEmailAddresses.html]: Need to keep IsShareEmailContactId conditions here

Comment (srosenbloom) [inline widgets/cmEmailAddresses.html]: Need to keep IsShareEmailContactId conditions here

File: widgets/cmEmailAddresses.html [hunk 3]
```diff
@@ -60,11 +60,10 @@
...
+                    <md-input-container md-no-float ng-click="toggleSharedEmail();">
                         <i class="fa cm-toggle-icon-small" ng-class="contact.sharedContact ? 'fa-toggle-on' : 'fa-toggle-off'">
                             <md-tooltip ng-if="!contact.IsShareEmailContactId && !contact.isNewContact && !contact.sharedContact">Click to use someone else's email address. (Please note that this will *replace* any existing emails for this contact.)</md-tooltip>
```
Comment (srosenbloom) [inline widgets/cmEmailAddresses.html]: Need to remove isNewContact conditions here

**Comment by rajush** [comment]
@srosenbloom  The new changes are there. Could you review the request? Thanks.

**Comment by rajush** [comment]
@srosenbloom Added extra validations.

**Pull request merged**